easybuild/tools/github.py

```diff
@@ -835,6 +880,153 @@ def is_patch_for(patch_name, ec):
+        return False
+
+    target = '%s/%s' % (pr_data['base']['repo']['owner']['login'], pr_data['base']['repo']['name'])
+    print_msg("Checking eligibility of %s PR #%s for merging..." % (target, pr_data['number']), prefix=False)
+
+    # check target branch, must be 'develop'
+    msg_tmpl = "* targets develop branch: %s"
+    if pr_data['base']['ref'] == 'develop':
+        print_msg(msg_tmpl % 'OK', prefix=False)
+    else:
+        res = not_eligible(msg_tmpl % "FAILED; found '%s'" % pr_data['base']['ref'])
+
+    # check test suite result, Travis must give green light
+    msg_tmpl = "* test suite passes: %s"
+    if pr_data['status_last_commit'] == 'success':
+        print_msg(msg_tmpl % 'OK', prefix=False)
+    elif pr_data['status_last_commit'] == 'pending':
+        res = not_eligible(msg_tmpl % "pending...")
+    elif pr_data['status_last_commit'] in ['error', 'failure']:
+        res = not_eligible(msg_tmpl % "FAILED")
+    else:
+        res = not_eligible(msg_tmpl % "(result unknown)")
+
+    # check for successful test report (checked in reverse order)
+    msg_tmpl = "* last test report is successful: %s"
+    test_report_regex = re.compile(r"^Test report by @\S+")
+    test_report_found = False
+    for comment in pr_data['issue_comments'][::-1]:
+        comment = comment['body']
+        if test_report_regex.search(comment):
+            if 'SUCCESS' in comment:
+                print_msg(msg_tmpl % 'OK', prefix=False)
+                test_report_found = True
+                break
+            elif 'FAILED' in comment:
+                res = not_eligible(msg_tmpl % 'FAILED')
+                test_report_found = True
+            else:
+                print_warning("Failed to determine outcome of test report for comment:\n%s" % comment)
+
+    if not test_report_found:
+        res = not_eligible(msg_tmpl % "(no test reports found)")
+
+    # check for approved review
+    approved_review_by = []
+    for review in pr_data['reviews']:
+        if review['state'] == 'APPROVED':
+            approved_review_by.append(review['user']['login'])
+
+    msg_tmpl = "* approved review: %s"
+    if approved_review_by:
+        print_msg(msg_tmpl % 'OK (by %s)' % ', '.join(approved_review_by), prefix=False)
+    else:
+        res = not_eligible(msg_tmpl % 'MISSING')
+
+    # check whether a milestone is set
+    msg_tmpl = "* milestone is set: %s"
+    if pr_data['milestone']:
+        print_msg(msg_tmpl % "OK (%s)" % pr_data['milestone']['title'], prefix=False)
+    else:
+        res = not_eligible(msg_tmpl % 'no milestone found')
+
+    return res
+
+
+def merge_pr(pr):
+    """
+    Merge specified pull request
+    """
+    github_user = build_option('github_user')
+    if github_user is None:
+        raise EasyBuildError("GitHub user must be specified to use --merge-pr")
+
+    pr_target_account = build_option('pr_target_account')
+    pr_target_repo = build_option('pr_target_repo')
+
```

Comment:
I'd even argue that for easyblocks a successful test report might be just as important as for the easyconfigs repo, given that both decide how the software is built. But we would have to make sure that the easyblock is used in the easyconfigs being built. In any case, I thought that at this moment the github integration worked just with the easyconfigs repo, so I wouldn't delay this PR because of this.